google/cloud/client/__init__.py

```diff
@@ -25,6 +25,7 @@
 import google.api_core.exceptions
 import google.auth
 from google.auth import environment_vars
+import google.auth.api_key
```

Comment:
Instead of bumping the minimum requirements , could we have the following

```
try:
    import google.auth.api_key
    HAS_GOOGLE_AUTH_API_KEY = True
except:
    HAS_GOOGLE_AUTH_API_KEY = False
``` 

We plan to drop `google-api-core` 1.x but it's better to decouple it from this feature PR. (and same for tests/unit/test_client.py)